freqtrade/rpc/rpc.py

```diff
@@ -537,6 +537,28 @@ def _rpc_forcebuy(self, pair: str, price: Optional[float]) -> Optional[Trade]:
         else:
             return None
 
+    def _rpc_delete(self, trade_id: str) -> Dict[str, str]:
+        """
+        Handler for delete <id>.
+        Delete the given trade
+        """
+        def _exec_delete(trade: Trade) -> None:
+            Trade.session.delete(trade)
+            Trade.session.flush()
+
+        with self._freqtrade._sell_lock:
+            trade = Trade.get_trades(
+                trade_filter=[Trade.id == trade_id, ]
+            ).first()
+            if not trade:
+                logger.warning('delete: Invalid argument received')
+                raise RPCException('invalid argument')
+
+            _exec_delete(trade)
+            Trade.session.flush()
+            self._freqtrade.wallets.update()
+            return {'result': f'Deleted trade {trade_id}.'}
+
```

Comment:
I don't think we can accept this part as is.

This works fine if you know exactly what state your trades and exchange account is in.
But in most cases when using telegram, you don't know the details (are there open orders ...)

There's things / checks we MUST have when deleting a trade:
* does it have open orders 
  * if yes - cancel these - eventually ignoring failures - assuming the order was cancelled on the exchange - but warning the user about it
  * this can have wider consequences - what if the order is half-filled? this could end up in unsellable assets (too low to place a new order)... 
* same for stoploss orders


obviously - since this is a crucial part - we will want this covered by tests. 
That's not an issue per se (i can help with tests if you're not familiar with pytest).